clang/lib/Parse/ParseTemplate.cpp

```diff
@@ -787,6 +787,23 @@ NamedDecl *Parser::ParseTemplateTemplateParameter(unsigned Depth,
                                                   unsigned Position) {
   assert(Tok.is(tok::kw_template) && "Expected 'template' keyword");
 
+  if (Token ahead = GetLookAheadToken(1);
+      ahead.isOneOf(tok::identifier, tok::ellipsis,
+                    tok::equal, tok::comma,
+                    tok::greater, tok::greatergreater, tok::greatergreatergreater)) {
```

Comment:
I think it might be simpler to just check if the next token is *not* `<` and correct to `typename` in that case since a template-template parameter always has to start with `template <` anyway.

Reply:
In hindsight yes I think thats a better solution. I believe this solution came about because I was just going to check for `identifier` but then realised there were more options for type templates and kept adding